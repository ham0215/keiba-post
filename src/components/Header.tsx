import React from 'react';
import styled from 'styled-components';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import { useRouter } from 'next/router';

const Header = styled.header`
  flex-grow: 1;
`;

const MenuButton = styled(IconButton)`
  margin-right: 2px;
`;

const Title = styled(Typography)`
  flex-grow: 1;
`;

export default function ButtonAppBar() {
  const router = useRouter();

  return (
    <Header>
      <AppBar position="static">
        <Toolbar>
          <MenuButton edge="start" color="inherit" aria-label="menu">
            <MenuIcon />
          </MenuButton>
          <Title variant="h6">
            <span onClick={() => router.push('/')}>Keiba Post</span>
          </Title>
          <Button color="inherit" onClick={() => router.push('/login')} > Login</Button>
        </Toolbar>
      </AppBar>
    </Header>
  );
}