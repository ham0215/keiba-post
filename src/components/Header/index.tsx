import React, { useContext } from 'react';
import styled from 'styled-components';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { useRouter } from 'next/router';
import Menu from './Menu';
import { UserContext } from '../../UserContext';
import UserAvatar from '../../components/UserAvatar'

const Header = styled.header`
  flex-grow: 1;
  background-color: green;
  color: #fff;
`;

const Title = styled(Typography)`
  flex-grow: 1;
`;

export default function ButtonAppBar() {
  const router = useRouter();

  const { currentUser } = useContext(UserContext);

  let loginButton;
  if (currentUser) {
    const { name, photoUrl } = currentUser
    loginButton = <UserAvatar name={name} photoUrl={photoUrl} />;
  } else {
    loginButton = (
      <Button color="inherit" onClick={() => router.push('/login')}>
        Login
      </Button>
    );
  }

  return (
    <Header>
      <AppBar position="static" color="transparent">
        <Toolbar>
          <Menu user={currentUser} />
          <Title variant="h6">
            <span onClick={() => router.push('/')}>Keiba Post</span>
          </Title>
          {loginButton}
        </Toolbar>
      </AppBar>
    </Header>
  );
}