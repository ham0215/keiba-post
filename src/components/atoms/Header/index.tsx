import styled from '@emotion/styled';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Link from 'components/atoms/Link';
import UserMenu from './UserMenu';

const Header = styled.header`
  flex-grow: 1;
  background-color: green;
  color: #fff;
`;

const Title = styled(Typography)`
  flex-grow: 1;
`;

export default function ButtonAppBar() {
  return (
    <Header>
      <AppBar position="static" color="transparent">
        <Toolbar>
          <Title variant="h6">
            <Link href="/">Keiba Post</Link>
          </Title>
          <UserMenu />
        </Toolbar>
      </AppBar>
    </Header>
  );
}
