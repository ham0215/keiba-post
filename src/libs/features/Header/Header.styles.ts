import styled from '@emotion/styled';
import Typography from '@mui/material/Typography';

export const Container = styled.header`
  flex-grow: 1;
  background-color: green;
  color: #fff;
`;

export const Title = styled(Typography)`
  flex-grow: 1;
`;
