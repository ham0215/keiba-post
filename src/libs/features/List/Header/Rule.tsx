import { useState } from 'react';
import Button from '@mui/material/Button';
import Modal from 'libs/ui/Modal';

export default function Rule() {
  const [open, setOpen] = useState(false);

  const handleOpen = () => {
    setOpen(true);
  };

  return (
    <>
      <Button variant="contained" onClick={handleOpen}>
        ルール
      </Button>
      <Modal open={open} setOpen={setOpen}>
        <div>
          【ルール】
          <br />
          ①1レース1,000円、一部のレースは3,000円
          <br />
          ②軸馬を決め、必ず軸馬を絡めて買う
          <br />
          ③払戻金額ではなく、回収率を競います
          <br />
          ④軸馬的中も競います。馬券当たらなくても、軸馬が3着以内で的中とします
          <br />
          <br />
          【期間】
          <br />
          ○上半期○
          <br />
          フェブラリーS～宝塚記念(12R)
          <br />
          ※3,000円のレース
          <br />
          天皇賞(春)、皐月賞、ダービー、宝塚記念
          <br />
          ○下半期○
          <br />
          スプリンターズS～有馬記念(12R)
          <br />
          ※3,000円のレース
          <br />
          菊花賞、天皇賞(秋)、ジャパンカップ、有馬記念
          <br />
          <br />
          【買い方の例】
          <br />
          ①番が軸の場合
          <br />
          例1.①複勝1,000円
          <br />
          例2.①-②③馬連各500円
          <br />
          例3.①-②③④⑤⑥三連複軸1頭流し各100円
          <br />
          例4.①-②③④三連単1着固定流し各100円 ＋ ①単勝400円
          <br />
          ×の例.①②③④⑤BOX
          <br />
        </div>
      </Modal>
    </>
  );
}
