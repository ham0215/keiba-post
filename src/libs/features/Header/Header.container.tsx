import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import { UserMenu } from './components/UserMenu';
import * as Styles from './Header.styles';
import { Link } from 'libs/ui/Link';

export function Header() {
  return (
    <Styles.Container>
      <AppBar position="static" color="transparent">
        <Toolbar>
          <Styles.Title variant="h6">
            <Link href="/">Keiba Post</Link>
          </Styles.Title>
          <UserMenu />
        </Toolbar>
      </AppBar>
    </Styles.Container>
  );
}
