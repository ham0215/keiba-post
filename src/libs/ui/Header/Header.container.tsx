import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Link from 'libs/ui/Link';
import { UserMenu } from './components/UserMenu';
import * as Styles from './Header.styles';

export function Header() {
  return (
    <Styles.Container>
      <AppBar position="static" color="transparent">
        <Toolbar>
          <Styles.Title variant="h6">
            <Link href="/">Keiba Post</Link>
          </Styles.Title>
          <UserMenu />
        </Toolbar>
      </AppBar>
    </Styles.Container>
  );
}
