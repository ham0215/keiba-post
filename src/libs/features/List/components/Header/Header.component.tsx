import FormControl from '@mui/material/FormControl';
import Grid from '@mui/material/Grid';
import InputLabel from '@mui/material/InputLabel';
import MenuItem from '@mui/material/MenuItem';
import Select from '@mui/material/Select';
import { RULE_2022_TAGS } from 'data/KeibaCalendar';
import { useRouter } from 'next/router';

import { Rule, Rule2022 } from '../Rule';
import * as Styles from './Header.styles';

export function Header({ tag }: { tag: string }) {
  const router = useRouter();

  return (
    <Styles.Container>
      <Grid container spacing={2} alignItems="center">
        <Grid item xs={6}>
          <FormControl fullWidth>
            <InputLabel id="selectTagLabel">年度</InputLabel>
            <Select
              labelId="selectTagLabel"
              id="selectTag"
              value={tag}
              label="selectTag"
              onChange={(e) => router.push(`/keiba?tag=${e.target.value}`)}
            >
              <MenuItem value="2023b">2023年後半</MenuItem>
              <MenuItem value="2023a">2023年前半</MenuItem>
              <MenuItem value="2022b">2022年後半</MenuItem>
              <MenuItem value="2022a">2022年前半</MenuItem>
              <MenuItem value="2021">2021年</MenuItem>
            </Select>
          </FormControl>
        </Grid>
        <Grid item xs alignItems="right">
          {RULE_2022_TAGS.some((v) => v === tag) ? <Rule2022 /> : <Rule />}
        </Grid>
      </Grid>
    </Styles.Container>
  );
}
