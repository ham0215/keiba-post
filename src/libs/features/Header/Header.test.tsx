import { render } from '@testing-library/react';

import { Header } from './Header.container';

// eslint-disable-next-line @typescript-eslint/no-require-imports
jest.mock('next/router', () => require('next-router-mock'));

describe('Header', () => {
  it('snapshot testing', () => {
    const { asFragment } = render(<Header currentUser={null} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
