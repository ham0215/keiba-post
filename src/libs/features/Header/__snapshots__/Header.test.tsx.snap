// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot testing 1`] = `
<DocumentFragment>
  <header
    class="css-19myip5"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-v526sl-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1g6ljil-MuiTypography-root"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Keiba Post
          </a>
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit MuiButton-root MuiButton-text MuiButton-textInherit MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorInherit css-1y942vo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Login
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </header>
</DocumentFragment>
`;
